src/vs/workbench/contrib/terminal/browser/terminalInstanceService.ts

```diff
@@ -17,27 +17,37 @@ import { Emitter, Event } from 'vs/base/common/event';
 import { TerminalContextKeys } from 'vs/workbench/contrib/terminal/common/terminalContextKey';
 import { Registry } from 'vs/platform/registry/common/platform';
 import { ILifecycleService, LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle';
+import { IWorkbenchEnvironmentService } from 'vs/workbench/services/environment/common/environmentService';
 
 export class TerminalInstanceService extends Disposable implements ITerminalInstanceService {
 	declare _serviceBrand: undefined;
 	private _terminalShellTypeContextKey: IContextKey<string>;
 	private _terminalInRunCommandPicker: IContextKey<boolean>;
 	private _terminalSuggestWidgetVisibleContextKey: IContextKey<boolean>;
 	private _configHelper: TerminalConfigHelper;
+	private _backendRegistration = new Map<string | undefined, { promise: Promise<void>; resolve?: () => void }>();
 
 	private readonly _onDidCreateInstance = new Emitter<ITerminalInstance>();
 	get onDidCreateInstance(): Event<ITerminalInstance> { return this._onDidCreateInstance.event; }
 
 	constructor(
 		@IInstantiationService private readonly _instantiationService: IInstantiationService,
 		@IContextKeyService private readonly _contextKeyService: IContextKeyService,
-		@ILifecycleService private readonly _lifecycleService: ILifecycleService
+		@ILifecycleService private readonly _lifecycleService: ILifecycleService,
+		@IWorkbenchEnvironmentService readonly _environmentService: IWorkbenchEnvironmentService,
 	) {
 		super();
 		this._terminalShellTypeContextKey = TerminalContextKeys.shellType.bindTo(this._contextKeyService);
 		this._terminalInRunCommandPicker = TerminalContextKeys.inTerminalRunCommandPicker.bindTo(this._contextKeyService);
 		this._terminalSuggestWidgetVisibleContextKey = TerminalContextKeys.suggestWidgetVisible.bindTo(this._contextKeyService);
 		this._configHelper = _instantiationService.createInstance(TerminalConfigHelper);
+
+
+		for (const remoteAuthority of [undefined, _environmentService.remoteAuthority]) {
+			let resolve: () => void;
+			const p = new Promise<void>(r => resolve = r);
+			this._backendRegistration.set(remoteAuthority, { promise: Promise.race([p, this._lifecycleService.when(LifecyclePhase.Eventually)]), resolve: resolve! });
```

Comment:
`RemoteTerminalBackendContribution` is registered to be created on `restored` phase but instantiation only occurs when idle, so we need to set an upper limit in this case `eventually` phase